src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -9731,93 +9560,302 @@ protected override bool Join(ref LocalState self, ref LocalState other)
                 return true;
             }
 
-            if (self.Capacity != other.Capacity)
-            {
-                Normalize(ref self);
-                Normalize(ref other);
-            }
+            Normalize(ref self);
+            Normalize(ref other);
 
             return self.Join(in other);
         }
 
+        internal sealed class LocalStateSnapshot
+        {
+            internal readonly int Id;
+            internal readonly LocalStateSnapshot? Container;
+            internal readonly BitVector State;
+
+            internal LocalStateSnapshot(int id, LocalStateSnapshot? container, BitVector state)
+            {
+                Id = id;
+                Container = container;
+                State = state;
+            }
+        }
+
         [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
-#if REFERENCE_STATE
-        internal class LocalState : ILocalDataFlowState
-#else
         internal struct LocalState : ILocalDataFlowState
-#endif
         {
+            private sealed class Boxed
```

Comment:
>Boxed [](start = 33, length = 5)

Nit: I think just `Box` would be a better name.

Reply:
We use the same name for `TypeWithAnnotations.Boxed`.

---
In reply to: [560562315](https://github.com/dotnet/roslyn/pull/50417#discussion_r560562315) [](ancestors = 560562315)